_pages/dev/ssr/server-side-rendering-optimization.md

```diff
@@ -86,53 +89,109 @@ The full list of parameters is described in the following section.
 
 You can configure the SSR optimization engine with a number of parameters, which are described as follows:
 
-- `timeout` is a number that indicates the amount of time (in milliseconds) during which the SSR server tries to render a page, before falling back to CSR. Once the delay has expired, the server returns the `index.html` of the CSR, which does not contain any pre-rendered markup. The CSR app (`index.html`) is served with a `Cache-Control:no-store` header. As a result, it is not stored by the cache layer. SSR pages do not contain this header because it is preferable to cache SSR pages.
+### timeout
+
+`Timeout` is a number that indicates the amount of time (in milliseconds) during which the SSR server tries to render a page, before falling back to CSR. Once the delay has expired, the server returns the `index.html` of the CSR, which does not contain any pre-rendered markup. The CSR app (`index.html`) is served with a `Cache-Control:no-store` header. As a result, it is not stored by the cache layer. SSR pages do not contain this header because it is preferable to cache SSR pages.
+
+In the background, the SSR server continues to render the SSR version of the page. Once this rendering finishes, the page is placed in a local cache to be returned the next time it is requested. By default, the server clears the page from its cache after returning it for the first time. _It is assumed and recommended that you are using an additional layer to cache pages externally (e.g. a CDN)_ (see [recommended SSR setup page](./server-side-rendering-setup.md)).
+
+A value of 0 will instantly return the CSR page.
+
+The default value is `3000` milliseconds.
+
+Recommendation: depending on the applications needs. Can be relatively big, assuming there is a proper caching layer strategy in place (e.g. CDN).
+
+### cache
+
+`cache` is a boolean that enables the built-in in-memory cache for pre-rendered URLs. This option is _not_ related to any kind of external caching layer (e.g. CDN). Even when this value is `false`, the cache is used to temporarily store the pages that finish rendering after the CSR fallback, so they can be served with the next request (after which, the cache is cleared). These temporarily cached responses are served with `cache-control: 'no-store'` HTTP header, in order to avoid caching them in an external caching layer.
+
+**Note:** The cache should be used carefully to avoid running out of memory. Using the `cacheSize` attribute can help avoid this.
```

Comment:
I think we need to emphasize that the "cache" is just server memory and it can cause memory leaks, running out of memory or server stalling.

Reply:
Nice, added.